crates/anvil-polkadot/tests/it/utils.rs

```diff
@@ -0,0 +1,161 @@
+use alloy_primitives::hex;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::{
+    api_server::{self, ApiHandle},
+    cmd::NodeArgs,
+    config::{AnvilNodeConfig, SubstrateNodeConfig},
+    opts::Anvil,
+    spawn,
+    substrate_node::service::Service,
+};
+use anvil_rpc::response::ResponseResult;
+use eyre::{Result, WrapErr};
+use futures::channel::oneshot;
+use parity_scale_codec::Decode;
+use polkadot_sdk::{
+    sc_cli::CliConfiguration,
+    sc_client_api::{BlockBackend, HeaderBackend},
+    sp_core::{storage::StorageKey, twox_128, H256},
+};
+use serde_json::{json, Value};
+
+pub struct TestNode {
+    pub service: Service,
+    pub api: ApiHandle,
+    pub _runtime_handle: tokio::runtime::Handle,
+}
+
+impl TestNode {
+    pub async fn new_custom<F>(anvil_args: Anvil, config_modifier: F) -> Result<Self>
+    where
+        F: FnOnce(&mut AnvilNodeConfig, &mut SubstrateNodeConfig),
```

Comment:
Is there something on AnvilNodeConfig that can't be directly configured via NodeArgs?

I don't think there should, so only the substrate config should be modifiable here